patches/minecraft/net/minecraft/item/ItemArmor.java.patch

```diff
@@ -1,6 +1,26 @@
 --- ../src-base/minecraft/net/minecraft/item/ItemArmor.java
 +++ ../src-work/minecraft/net/minecraft/item/ItemArmor.java
-@@ -221,6 +221,19 @@
+@@ -30,6 +30,8 @@
+ import net.minecraftforge.fml.relauncher.Side;
+ import net.minecraftforge.fml.relauncher.SideOnly;
+ 
++import javax.annotation.Nullable;
++
+ public class ItemArmor extends Item
+ {
+     private static final int[] field_77882_bY = new int[] {13, 15, 16, 11};
+@@ -187,7 +189,9 @@
+ 
+     public boolean func_82789_a(ItemStack p_82789_1_, ItemStack p_82789_2_)
+     {
+-        return this.field_77878_bZ.func_151685_b() == p_82789_2_.func_77973_b() ? true : super.func_82789_a(p_82789_1_, p_82789_2_);
++        ItemStack mat = this.field_77878_bZ.getRepairItemStack();
++        if (!mat.func_190926_b() && net.minecraftforge.oredict.OreDictionary.itemMatches(mat,p_82789_2_,false)) return true;
++        return super.func_82789_a(p_82789_1_, p_82789_2_);
```

Comment:
You should make this 1 line like the vanilla version to make the patch small. 
`OreDictionary.itemMatches` already checks `!mat.func_190926_b()` so you don't need that check here.